core/domain/exp_domain.py

```diff
@@ -1224,6 +1224,46 @@ def create_default_interaction(cls, default_dest_state_name):
             cls._DEFAULT_INTERACTION_ID, {}, [], default_outcome, [], [], {})
 
 
+    def get_all_html_content_strings(self):
+        """Get all html content strings in the interaction.
+
+        Returns:
+            list(str): The list of all html content strings in the interaction.
+        """
+        html_list = []
+
+        for answer_group in self.answer_groups:
+            outcome_html = answer_group.outcome.feedback.html
+            html_list = html_list + [outcome_html]
+
+        if self.id == 'ItemSelectionInput':
+            for answer_group in self.answer_groups:
+                for rule_spec in answer_group.rule_specs:
+                    rule_spec_html = rule_spec.inputs['x']
```

Comment:
This seems a bit odd. What do you get from this that you don't already get from the interaction's customization args?

Reply:
Yes, it provides the same data, but since the function returns all html values, I included this as well though it is not needed for validation as same error will be generated,  Shall I will remove this?

Reply:
Oh, I see, good point. I think it's fine to keep it. Just add a comment here saying: "Note that ItemSelectionInput replicates the customization arg HTML in its answer groups."

Reply:
Done!